### DISCUSSION TITLE
Trying to use DNS for Node Name

### DISCUSSION THREAD

**Discussion by kphaneuf** [Esri/arcgis-powershell-dsc#509] [Q&A] [answered]
We want to use a DNS for node name because we don't want the machine name propagated throughout the admin URLs and server URLs in enterprise. When using the DNS we get the following error:
"WinRM cannot process the request. The following error occurred while using Kerberos authentication: Cannot find the computer <FQDN> Verify that the computer exists on the network and that the name provided is spelled correctly."

I see in the Wiki variables that only machine name or IP address can be used for Node Name. Is it possible to use DNS? How would we get around the machine name being used for all the admin URLs?

**Comment by cameronkroeker** [comment]
Hi @kphaneuf,

The ArcGIS Module uses WinRM HTTP to connect to each target node, from the orchestrating node, to run the configuration. Based on the error it looks like WinRM isn't able to connect via the DNS Alias. Some suggestions:

- Add the DNS alias to the trusted hosts list on the orchestrating node, and all target nodes. For example, launch PowerShell as an Administrator, and run:
  -  `winrm s winrm/config/client '@{TrustedHosts="alias.domain.com, alias2.domain.com"}'`
- Ensure firewall allows incoming connections for the WinRM HTTP port 5985
- Check to see if you can connect from the orchestrating machine to the target nodes without using the ArcGIS Module. For example, launch PowerShell as an Administrator, and run these 2 commands:
   - `Test-WSMan -ComputerName alias.domain.com -Authentication default -Port 5985`
   - `New-CimSession -Port 5985 -ComputerName alias.domain.com -Authentication default`

Thanks,
Cameron K.

**Reply by kphaneuf** [reply]
Thanks @cameronkroeker. I have added the DNS alias as a trusted host, added the firewall inbound rule, and we still are seeing the same error when we run the Test-WSMan command.  When we run New-CimSession it just shows the ID, Name, InstanceID, ComputerName, and Protocol, no errors.

**Reply by kphaneuf** [reply]
Also, the Test-WSMan command happens when I use the DNS for the machine that is running the command as well as a separate machine.

**Reply by kphaneuf** [reply, marked as answer]
@cameronkroeker We found that it was actually an SPN issue.  We hadn't set the spn on either of the servers. Once we did that it fixed it and Test-WSMan works.

**Reply by cameronkroeker** [reply]
@kphaneuf Glad to hear that setting the SPN on the servers resolved the issue!
